services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -523,22 +524,23 @@ async def delete_file(
                 if not can.delete:
                     raise FileAccessRightError(access_right="delete", file_id=file_id)
 
-        with suppress(FileMetaDataNotFoundError):
-            # NOTE: deleting might be slow, so better ensure we release the connection
-            async with self.engine.acquire() as conn:
-                file: FileMetaDataAtDB = await db_file_meta_data.get(
-                    conn, TypeAdapter(SimcoreS3FileID).validate_python(file_id)
+        await get_s3_client(self.app).delete_objects_recursively(
+            bucket=self.simcore_bucket_name,
+            prefix=file_id,
+        )
+
+        async with self.engine.acquire() as conn:
+            file_fmd = await get_fmd(conn, file_id)
```

Comment:
I would propose to call this: `try_get_fmd` since it does not raise.
and also you can do:
```python
if file_fmd := await try_get_fmd(conn, file_id):
    await db_file_meta_data.delete(conn, [file_id])
if parent_director_file_id := await get_directory_file_id(conn, file_id):
    await db_file_meta_data.upsert(conn, parent_director_file_id)